### PR TITLE
fix: `is_duplicated` was returning wrong-length result for PyArrow and Dask

### LINKED ISSUE #1678: bug: `is_duplicated` returns wrong number of elements for pyarrow
```python
import pandas as pd
import polars as pl
import pyarrow as pa
import narwhals as nw

df = pa.table({'a': [1, None, 1, None, 2, 3]})
print(nw.from_native(df).select(nw.col('a').is_duplicated()).to_native())
df = pl.DataFrame({'a': [1, None, 1, None, 2, 3]})
print(nw.from_native(df).select(nw.col('a').is_duplicated()).to_native())
df = pd.DataFrame({'a': [1, None, 1, None, 2, 3]})
print(nw.from_native(df).select(nw.col('a').is_duplicated()).to_native())
```

```
pyarrow.Table
a: bool
----
a: [[true,true,false,false]]
shape: (6, 1)
┌───────┐
│ a     │
│ ---   │
│ bool  │
╞═══════╡
│ true  │
│ true  │
│ true  │
│ true  │
│ false │
│ false │
└───────┘
       a
0   True
1   True
2   True
3   True
4  False
5  False
```

The last one has fewer elements

### PR DESCRIPTION
**Pull request by MarcoGorelli**
closes #1678 

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #\<issue number\>
- Closes #\<issue number\>

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

File: narwhals/_arrow/dataframe.py [hunk 1]
```diff
@@ -602,31 +602,35 @@ def is_duplicated(self: Self) -> ArrowSeries:
...
-        df = self._native_frame
+        index_token = generate_temporary_column_name(n_bytes=8, columns=self.columns)
+        df = self.with_row_index(index_token)._native_frame
 
         columns = self.columns
         col_token = generate_temporary_column_name(n_bytes=8, columns=columns)
         row_count = (
             df.append_column(col_token, pa.repeat(pa.scalar(1), len(self)))
-            .group_by(columns)
+            .group_by([x for x in columns if x != index_token])
```
Comment (FBruzzesi) [inline narwhals/_arrow/dataframe.py:612]: `columns` in line 608 should still be the original frame columns as it is from `self` and not from the newly created `df`

File: narwhals/_arrow/dataframe.py [hunk 2]
```diff
@@ -602,31 +602,35 @@ def is_duplicated(self: Self) -> ArrowSeries:
 
         from narwhals._arrow.series import ArrowSeries
 
-        df = self._native_frame
+        index_token = generate_temporary_column_name(n_bytes=8, columns=self.columns)
+        df = self.with_row_index(index_token)._native_frame
 
         columns = self.columns
         col_token = generate_temporary_column_name(n_bytes=8, columns=columns)
```
Comment (FBruzzesi) [inline narwhals/_arrow/dataframe.py:609]: Very unlikely, but `columns` here does not include `index_token` (for the comment below)

**Review by FBruzzesi — COMMENTED** [review]
Thanks @MarcoGorelli , great spotting and fixing this issue! (It was probably me who implemented it incorrectly 🙈)

I left a couple on comments in the pyarrow codebase, both regarding the use of `columns` object

**Comment by MarcoGorelli** [comment]
awesome, thanks @FBruzzesi for your extra commit and review!

**Pull request merged**